Return proper error in etcd healthcheck

alena1108 opened the pull request:
https://github.com/rancher/rke/issues/1300

alena1108 commented:
@superseb reworked the following way:

* Only log in debug in isEtcdHealthy util method as we don't want to log in warn for every iteration. Return final error if all iterations fail
* Log error in Warn in ReloadEtcdCluster and RunEtcdPlane for a particular host failure. So you can see which host check resulted in error
* Return a generic error from ReloadEtcdCluster and RunEtcdPlane if *all* iterations fail. Let the caller stack to log it in error (or fatal). Remove redundant [etcd] from the returned error

superseb commented:
Normal with one down etcd node but cluster still healthy:

```
INFO[0034] [etcd] Successfully started etcd plane.. Checking etcd cluster health                         
WARN[0052] [etcd] host [54.246.149.88] failed to check etcd health: failed to get /health for host [54.246.149.88]: Get https://172.26.7.200:2379/health: Unable to access the service on 172.26.7.200:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused)                
```

Debug with one down etcd node but cluster still healthy:

```
INFO[0034] [etcd] Successfully started etcd plane.. Checking etcd cluster health                                                                                                                                                                                                                                              
DEBU[0034] [etcd] check etcd cluster health                                                                                                                                                                                                                                                                                   
DEBU[0035] <nil>                                                                                                                                                                                                                                                                                                              
DEBU[0041] <nil>                                                                                                                                                                                                                                                                                                              
DEBU[0047] <nil>                                                                                                                                                                                                                                                                                                              
WARN[0052] [etcd] host [54.246.149.88] failed to check etcd health: failed to get /health for host [54.246.149.88]: Get https://172.26.7.200:2379/health: Unable to access the service on 172.26.7.200:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused)                
DEBU[0052] [etcd] check etcd cluster health                                                                                                                                                                                                                                                                                   
DEBU[0053] [etcd] etcd host [34.242.232.151] reported healthy=true                                                                  
```

Normal with broken cluster:

```
INFO[0035] [etcd] Successfully started etcd plane.. Checking etcd cluster health                                                                                                                                                                                                                                              
WARN[0053] [etcd] host [54.246.149.88] failed to check etcd health: failed to get /health for host [54.246.149.88]: Get https://172.26.7.200:2379/health: Unable to access the service on 172.26.7.200:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused)                
WARN[0071] [etcd] host [34.242.232.151] failed to check etcd health: failed to get /health for host [34.242.232.151]: Get https://172.26.4.176:2379/health: Unable to access the service on 172.26.4.176:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused)              
WARN[0074] [etcd] host [54.171.8.251] reported healthy=false                                                                                                                                                                                                                                                                  
FATA[0074] [etcd] Failed to bring up Etcd Plane: etcd cluster is unhealthy; check log for more details
```

Debug with broken cluster:

```
INFO[0035] [etcd] Successfully started etcd plane.. Checking etcd cluster health 
DEBU[0035] [etcd] check etcd cluster health             
DEBU[0036] <nil>                                        
DEBU[0042] <nil>                                        
DEBU[0047] <nil>                                        
WARN[0052] [etcd] host [54.246.149.88] failed to check etcd health: failed to get /health for host [54.246.149.88]: Get https://172.26.7.200:2379/health: Unable to access the service on 172.26.7.200:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused) 
DEBU[0052] [etcd] check etcd cluster health             
DEBU[0053] <nil>                                        
DEBU[0059] <nil>                                        
DEBU[0065] <nil>                                        
WARN[0070] [etcd] host [34.242.232.151] failed to check etcd health: failed to get /health for host [34.242.232.151]: Get https://172.26.4.176:2379/health: Unable to access the service on 172.26.4.176:2379. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused) 
DEBU[0070] [etcd] check etcd cluster health             
DEBU[0071] [etcd] etcd host [54.171.8.251] reported healthy=false 
DEBU[0072] [etcd] etcd host [54.171.8.251] reported healthy=false 
DEBU[0073] [etcd] etcd host [54.171.8.251] reported healthy=false 
WARN[0073] [etcd] host [54.171.8.251] reported healthy=false 
FATA[0073] [etcd] Failed to bring up Etcd Plane: etcd cluster is unhealthy; check log for more details 
```

* nil prints should be the errors for the host
* `check log for more details` is not informative enough for the user, we need to specify what log they need to check and if possible, specify what hosts failed to report healthy (similar to port checker, where we specify what host cant connect to what host)

alena1108 commented:
> nil prints should be the errors for the host

Fixed with https://github.com/rancher/rke/pull/1735/files#diff-7aec623fe39b89b63863840e34fcb58bR66

> check log for more details is not informative enough for the user, we need to specify what log they need to check and if possible, specify what hosts failed to report healthy (similar to port checker, where we specify what host cant connect to what host)

Fixed with https://github.com/rancher/rke/pull/1735/files#diff-ec815150e8926f2182c0d2d116d3ca3dR256
